src/werkzeug/wsgi.py

```diff
@@ -150,25 +108,12 @@ def get_host(
     :raise ~werkzeug.exceptions.SecurityError: If the host is not
         trusted.
     """
-    if "HTTP_HOST" in environ:
-        rv = environ["HTTP_HOST"]
-        if environ["wsgi.url_scheme"] == "http" and rv.endswith(":80"):
-            rv = rv[:-3]
-        elif environ["wsgi.url_scheme"] == "https" and rv.endswith(":443"):
-            rv = rv[:-4]
-    else:
-        rv = environ["SERVER_NAME"]
-        if (environ["wsgi.url_scheme"], environ["SERVER_PORT"]) not in (
-            ("https", "443"),
-            ("http", "80"),
-        ):
-            rv += f":{environ['SERVER_PORT']}"
-    if trusted_hosts is not None:
-        if not host_is_trusted(rv, trusted_hosts):
-            from .exceptions import SecurityError
-
-            raise SecurityError(f'Host "{rv}" is not trusted')
-    return rv
+    return sansio_get_host(
+        environ["wsgi.url_scheme"],
+        environ.get("HTTP_HOST"),
+        (environ["SERVER_NAME"], int(environ["SERVER_PORT"])),
```

Comment:
I see that this matches the old behavior for `port`, but it should probably match the new `try/except` behavior in `Request.__init__`. Might also make sense to add a `_get_server()` (private) helper to `wsgi.py` and reuse it in both places. I'll add this along with some docs cleanup.